src/backend/distributed/commands/multi_copy.c

```diff
@@ -1751,6 +1752,12 @@ CitusCopyDestReceiverStartup(DestReceiver *dest, int operation,
 	LockShardListMetadata(shardIntervalList, ShareLock);
 	LockShardListResources(shardIntervalList, ShareLock);
 
+	/*
+	 * We might have some concurrent metadata changes and we need to accept the cache
+	 * invalidation messages to learn about the changes.
+	 */
+	AcceptInvalidationMessages();
```

Comment:
Hmm, we're planning to backport this PR. So, I'd be in favor of discussing that later. But your suggestion seems a pretty useful approach if `AcceptInvalidationMessages ()` is a cheap function to call.

(For reloading the cache entry, I was also thinking to add to make the code more readable. Also, we're setting `copyDest->tableMetatada = cacheEntry` just 1 line below. So, it makes sense to trigger the cache invalidation there.)